File: ArraysAndHashing/ContainsDuplicate.py
''' 
Given an integer array nums, return true if any value appears more than once in 
the array, otherwise return false.
'''

def hasDuplicate(self, nums: List[int]) -> bool:
    seen = set()
    for i in nums:
        if i in seen:
            return True
        else:
            seen.add(i)
    return False
         

File: python/ContainsDuplicate.cpp
#include <vector>
#include <unordered_set>

class Solution {
public:
    bool hasDuplicate(std::vector<int>& nums) {
        std::unordered_set<int> seen;
        for (int n : nums) {
            if (seen.count(n)) {
                return true;
            }
            seen.insert(n);
        }
        return false;
    }
};